llvm/include/llvm/ADT/APFloat.h

```diff
@@ -773,13 +791,362 @@ hash_code hash_value(const DoubleAPFloat &Arg);
 DoubleAPFloat scalbn(const DoubleAPFloat &Arg, int Exp, IEEEFloat::roundingMode RM);
 DoubleAPFloat frexp(const DoubleAPFloat &X, int &Exp, IEEEFloat::roundingMode);
 
+class DFPFloat final : public APFloatBase {
+public:
+  /// \name Constructors
+  /// @{
+
+  DFPFloat(const fltSemantics &) {} // Default construct to +0.0
+  DFPFloat(const fltSemantics &, integerPart) { assert(false && "Not Implemented"); }
+  DFPFloat(const fltSemantics &, uninitializedTag) { assert(false && "Not Implemented"); }
+  DFPFloat(const fltSemantics &, const APInt &) { assert(false && "Not Implemented"); }
+  explicit DFPFloat(double d) { assert(false && "Not Implemented"); }
+  explicit DFPFloat(float f) { assert(false && "Not Implemented"); }
```

Comment:
Tentatively going to say we shouldn't have these constructors at all. For DFP, an easy decimal string literal conversion is probably more useful than double or float (after all, you wouldn't want to have rounding errors trying to convert `0.1` to decimal32!)